packages/@react-spectrum/combobox/src/MobileComboBox.tsx

```diff
@@ -159,9 +186,28 @@ const ComboBoxButton = React.forwardRef(function ComboBoxButton(props: ComboBoxB
       valueId,
       validationState === 'invalid' ? invalidId : null
     ].filter(Boolean).join(' '),
-    elementType: 'div'
+    elementType: 'div',
+    // Prevent focus on press and focus on press end to avoid situations where user
+    // presses and holds on the combobox button and tray immediately opens due to menuTrigger=focus
+    // @ts-ignore
+    preventFocusOnPress: true,
+    onPressEnd: () => {
+      focusSafely(ref.current);
+    }
   }, ref);
 
+  // Prevent focus event when pressing down and holding on combobox button. This is to stop tray from opening until user releases the button.
+  // Prevents tray open -> useInteractOutside called on press end -> tray closing immediately when menuTrigger = "focus" and user presses and holds combobox button on mobile
+  useEffect(() => {
+    let button = ref.current;
+    let onTouchStart = (e) => e.preventDefault();
+    button.addEventListener('touchstart', onTouchStart);
+
+    return () => {
+      button.removeEventListener('touchstart', onTouchStart);
+    };
+  }, [ref]);
```

Comment:
what's the difference between this and the preventFocusOnPress?

Reply:
So we don't actually call preventDefault on touchStart in usePress, removed by this pull in favor of an alternative method for iOS: https://github.com/adobe/react-spectrum/pull/317. However, Android seems to work differently where a long press causes the button to be focused, hence the need to prevent default on touchStart here. I'll add this to the comment